concordium-consensus/tests/consensus/ConcordiumTests/KonsensusV1/TransactionProcessingTest.hs

```diff
@@ -347,9 +349,81 @@ dummyTransaction =
             }
     payload = encodePayload $ Transfer dummyAccountAddress 10
 
--- | The block item for 'dummyTransaction'.
-dummyTransactionBI :: BlockItem
-dummyTransactionBI = normalTransaction dummyTransaction
+-- | A dummy update instruction.
+--
+-- TODO (drsk) The signature needs to be changed to a correct one.
+dummyUpdateInstruction :: TransactionTime -> WithMetadata UpdateInstruction
+dummyUpdateInstruction effTime =
+    addMetadata ChainUpdate 0 $
+        UpdateInstruction
+            { uiHeader = hdr,
+              uiPayload = cplt,
+              uiSignHash = makeUpdateInstructionSignHash "dummy",
+              uiSignatures = UpdateInstructionSignatures{signatures = Map.singleton 0 sig}
+            }
+  where
+    sig = SigScheme.sign dummySigSchemeKeys "updateInstruction"
+    hdr =
+        UpdateHeader
+            { updateSeqNumber = 1,
+              updateEffectiveTime = effTime,
+              updateTimeout = 2,
+              updatePayloadSize = 100
+            }
+    cplt =
+        CreatePLTUpdatePayload $
+            CreatePLT
+                { _cpltTokenId = TokenId "dummyToken",
+                  _cpltTokenModule = TokenModuleRef $ Hash.hash "dummyToken",
+                  _cpltDecimals = 4,
+                  _cpltInitializationParameters = TokenParameter ""
+                }
+
+-- | The block item for 'dummyNormalTransaction'.
+dummyNormalTransactionBI :: BlockItem
+dummyNormalTransactionBI = normalTransaction dummyNormalTransaction
+
+-- | The block item for 'dummyUpdateInstruction'.
+dummyChainUpdateBI :: TransactionTime -> BlockItem
+dummyChainUpdateBI effTime = chainUpdate $ dummyUpdateInstruction effTime
+
+-- | Test for transaction verification
+testTransactionVerification ::
+    forall pv.
+    (IsConsensusV1 pv, IsProtocolVersion pv) =>
+    SProtocolVersion pv ->
+    Spec
+testTransactionVerification _ = describe "transaction verification" $ do
```

Comment:
Does this need to be guarded for protocol versions >= P9?

Reply:
It's now guarded. I added the guard at the call side to test only for protocols >= P9. Not sure if I could also add a constraint to the type of the test.